Dev ex01 dnakano

dai65527 opened the pull request:
TOYASANメモ用です。

dai65527 committed:
Ex01: Add: just vue create but in ts

- 以下の設定で`vue create pokedex` しました。
- TSのテンプレートで作ってます
- オプションは適当です。（とりあえず）
- 触ってみて変更する可能性あり
? Please pick a preset: Manually select features
? Check the features needed for your project: Choose Vue version, Babel, TS, Linter
? Choose a version of Vue.js that you want to start the project with 2.x
? Use class-style component syntax? Yes
? Use Babel alongside TypeScript (required for modern mode, auto-detected polyfills, transpilin
g JSX)? Yes
? Pick a linter / formatter config: Prettier
? Pick additional lint features: Lint on save
? Where do you prefer placing config for Babel, ESLint, etc.? In package.json
? Save this as a preset for future projects? No

dai65527 committed:
Ex01: Change: Recreate pokedex

- やはり`Use class-style component syntax?` は Noの方がTS初心者には優しいので、そうしました。

dai65527 committed:
Ex01: Add: pokemon.ts

- has class and interface contains pokemon information
- id, name and imageUrl

dai65527 committed:
Ex02: Fixed: Re-remake by class style syntax

```
? Please pick a preset: Manually select features
? Check the features needed for your project: Choose Vue version, Babel, TS, Lin
ter
? Choose a version of Vue.js that you want to start the project with 2.x
? Use class-style component syntax? Yes
? Use Babel alongside TypeScript (required for modern mode, auto-detected polyfi
lls, transpiling JSX)? Yes
? Pick a linter / formatter config: Prettier
? Pick additional lint features: Lint on save
? Where do you prefer placing config for Babel, ESLint, etc.? In dedicated confi
g files
? Save this as a preset for future projects? No
```

dai65527 committed:
Ex01: Add: structure of Components

Added following
- Pokedex component
- PokeList component
- PokeListItem component
- Pokemon class

dai65527 committed:
Ex01: Update: get pokemons as array from function

dai65527 committed:
Ex01: Update: fetch data from pokeapi using axios

- install axios
- add fetchPokemons function to pokemon.ts
- modify on PokeList vue to use fetchPokemons function
- slight modification on Pokedex.vue (adjust to prettier)

dai65527 committed:
Ex01: Update: add infinite scroll

- npm vue-infinite-lodingした
- 全てのデータが読み出せた
- 謎に中国語が混じり出す

dai65527 committed:
Ex01: Update: Vuetifyでちょっと整えました

dai65527 committed:
Ex01: Fix: include path of pokemon.ts

ToYeah commented on ex01/pokedex/README.md:
````diff
@@ -0,0 +1,24 @@
+# pokedex
+
+## Project setup
+```
+npm install
+```
+
+### Compiles and hot-reloads for development
+```
+npm run serve
+```
+
+### Compiles and minifies for production
+```
+npm run build
+```
+
+### Lints and fixes files
+```
+npm run lint
+```
+
+### Customize configuration
+See [Configuration Reference](https://cli.vuejs.org/config/).
````
README消しましょうよ

dai65527 replied:
だが消さぬ。

ToYeah commented on ex01/pokedex/src/App.vue:
```diff
@@ -0,0 +1,19 @@
+<template>
+  <div id="app">
+    <v-app>
+      <Pokedex />
+    </v-app>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from "vue-property-decorator";
+import Pokedex from "./components/Pokedex.vue";
```
v-appを消す
langをjsにする

dai65527 replied:
直しました！ JSでも問題なく動きます！

ToYeah commented on ex01/pokedex/src/components/PokeList.vue:
```diff
@@ -0,0 +1,58 @@
+      />
+    </div>
+    <InfiniteLoading
+      v-if="!flgErrLoading || !flgErrLoading"
```
フラグ多いです

dai65527 replied:
直しました！

ToYeah commented on ex01/pokedex/src/models/pokemon.ts:
```diff
@@ -0,0 +1,64 @@
+const POKEAPI_ROOT = "https://pokeapi.co/api/v2/";
+
+export default class Pokemon {
+  private id_: number;
+  name_: string;
+  imageUrl_: string;
```
privateにしましょう

dai65527 replied:
すみませんでした。

ToYeah commented on ex01/pokedex/src/models/pokemon.ts:
```diff
@@ -0,0 +1,64 @@
+    POKEAPI_ROOT + `pokemon/?offset=${offset}&limit=${limit}`
+  );
+  const pokemons: Pokemon[] = await Promise.all(
+    resPokeList.data.results.map(async (result: { url: string }) => {
```
返り値の型をつけたほうが安全だと思います。

dai65527 replied:
付けました！

ToYeah commented on ex01/pokedex/src/models/pokemon.ts:
```diff
@@ -0,0 +1,64 @@
+  const resPokeSpecies = await axios.get(resPokeInfo.data.species.url);
+  return new Pokemon(
+    resPokeInfo.data.id,
+    resPokeSpecies.data.names[9].name,
```
中国語でるので、日本語名探した方がいいと思います！

dai65527 replied:
イケてる関数に変更しました

ToYeah reviewed (changes requested):
直したらUIやります！

dai65527 committed:
Ex01: Fix: App.vue as comment by ToYeah

- TS to JS
- Move v-app tag to Pokedex.vue
- App.vue is now same as subject

dai65527 committed:
Ex01: Fix: 読み込み周辺修正

- flgErrLoadingが余計にあったとこ消しました。
- 読み込み最大を898に設定
- Axiosでエラーが出た際にエラーメッセージを追加。（内容とデザインは暫定）

dai65527 committed:
Ex01: Fixed: 型定義とprivate

- 型定義抜けてたところ追加
- Pokemon classのprivate属性抜けていたところを追加

dai65527 committed:
Ex01: fix: 言語指定

- イケてる関数に変更
- PokeListコンポーネントから言語を指定できるようにしました！

dai65527 commented:
イケてる関数に変更したのでご確認お願いします！
